src/main/java/org/baldogru/imagegallery/service/validator/ValidationSummaryInterpreterImpl.java

```diff
@@ -10,11 +12,12 @@
 
     @Override
     public void processValidationSummary(ValidationSummary validationSummary) {
-        //Możesz sprobowac cos zaimplementowac.
-        //Chodzi np. o to, ze jak wynik walidacji bedzie chujowy,
-        //to zeby to np rzuciło na razie wyjatkiem,
-        //a jak bedzie ok to zeby po prostu nie robilo nic.
-        //Potem wymyslimy jak tu zaimplementować fajną obsługę co zwróci ładny komunikat z listą błędów.
-        //Chyba, że masz pomysł, to możesz coś zasugerować, ale wtedy raczej tutaj nie będzie zwracany void :)
+        throwExceptionIfValidationSummaryIsFailed(validationSummary);
+    }
+
+    private void throwExceptionIfValidationSummaryIsFailed(ValidationSummary validationSummary) {
+        if (validationSummary.getValidationStatus().equals(ValidationStatus.FAIL)) {
```

Comment:
Jeżeli w tej kolejności używasz equals to musisz się wcześniej upewnić, że nie będziesz mieć nulla. W przeciwnym razie bezpieczniej jest zrobić equals na czymś co jest pewniakiem, czyli tutaj jest to ValidationStatus.FAIL

Reply:
Nie za bardzo rozumiem, w jaki sposób "W przeciwnym razie bezpieczniej jest zrobić equals na czymś co jest pewniakiem, czyli tutaj jest to ValidationStatus.FAIL"
miałbym to porównać ?